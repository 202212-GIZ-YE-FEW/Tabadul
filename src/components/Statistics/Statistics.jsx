import Image from "next/image";
import React from "react";

import {
    IntroPhoto,
    StaticInnerImage,
    Statistic,
    StatisticHeading,
    StatisticParagraph,
    StatisticsContainer,
    StatisticsParant,
} from "./Statistics.style";
import StatisticImagetThired from "../../assets/Image/icon_bar_chart.svg";
import StatisticImageSecond from "../../assets/Image/icon_pie_chart.svg";
import StatisticImage from "../../assets/Image/pyramid_chart.svg";
import StatisticsContainerImage from "../../assets/Image/StatisticsContainer.svg";
import StatisticImageFifth from "../../assets/Image/Vector.svg";

function Statistics() {
    return (
        <StatisticsContainer>
            <IntroPhoto
                src={StatisticsContainerImage}
                alt='StatisticsContainerImage'
                width={1006.7}
                height={464.69}
                priority={true}
            />
            <StatisticsParant>
                <Statistic>
                    <StaticInnerImage
                        src={StatisticImage}
                        alt='StatisticImage'
                        width={78.24}
                        height={78.24}
                    />
                    <StatisticHeading>1/3</StatisticHeading>
                    <StatisticParagraph>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore
                        magna aliqua.
                    </StatisticParagraph>
                </Statistic>
                <Statistic>
                    <Image
                        src={StatisticImageSecond}
                        alt='StatisticImage'
                        width={78.24}
                        height={78.24}
                    />
                    <StatisticHeading>75%</StatisticHeading>
                    <StatisticParagraph>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore
                        magna aliqua.
                    </StatisticParagraph>
                </Statistic>
                <Statistic>
                    <Image
                        src={StatisticImagetThired}
                        alt='StatisticImage'
                        width={78.24}
                        height={78.24}
                    />
                    <StatisticHeading>312</StatisticHeading>
                    <StatisticParagraph>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore
                        magna aliqua.
                    </StatisticParagraph>
                </Statistic>
                <Statistic>
                    <Image
                        src={StatisticImageFifth}
                        alt='StatisticImage'
                        width={78.24}
                        height={78.24}
                    />
                    <StatisticHeading>Fact</StatisticHeading>
                    <StatisticParagraph>
                        Lorem ipsum dolor sit amet, consectetur adipiscing elit,
                        sed do eiusmod tempor incididunt ut labore et dolore
                        magna aliqua.
                    </StatisticParagraph>
                </Statistic>
            </StatisticsParant>
        </StatisticsContainer>
    );
}

export default Statistics;
