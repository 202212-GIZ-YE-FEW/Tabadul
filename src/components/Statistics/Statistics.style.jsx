import styled from "@emotion/styled";
import Image from "next/image";

export const StatisticsContainer = styled.div`
    height: 658px;
    background-color: #33956d;
    position: relative;
    z-index: -1;
    display: grid;
    place-items: center;
    font-family: Poppins;
    &::after {
        content: "";
        position: absolute;
        left: 0;
        top: 0;
        width: 100%;
        height: 100%;
        background-color: rgb(51 149 109 / 80%);
        z-index: 200;
    }
    @media screen and (max-width: 768px) {
        height: auto;
    }
`;
export const IntroPhoto = styled(Image)`
    width: 100%;
    max-width: 100%;
    margin: 0 auto;
    @media screen and (max-width: 768px) {
        height: 280px;
    }
`;
export const StatisticsParant = styled.div`
    position: relative;
    top: -130px;
    width: 90%;
    max-width: 1689px;
    height: auto;
    background-color: #fff;
    box-shadow: 0px 8px 24px rgba(0, 0, 0, 0.25);
    border-radius: 45px;
    margin: auto;
    z-index: 500;
    padding: 30px 20px;
    display: flex;
    flex-direction: row;
    justify-content: center;
    gap: 44px;
    flex-wrap: wrap;
`;

export const Statistic = styled.div`
    text-align: center;
    padding-top: 41px;
    width: 363.31px;
`;

export const StaticInnerImage = styled(Image)`
    @media screen and (max-width: 768px) {
        width: 71.31px;
        height: 62.4px;
    }
`;

export const StatisticHeading = styled.h2`
    font-size: 30px;
    font-weight: 700;
    color: #4e5769;
    margin-top: 20px;
    margin-bottom: 20px;
    @media screen and (max-width: 768px) {
        margin-top: 11px;
        margin-bottom: 11px;
    }
`;
export const StatisticParagraph = styled.p`
    font-size: 24px;
    color: #000000;
    line-height: 36px;
    font-weight: 400;
    @media screen and (max-width: 768px) {
        font-size: 20px;
        line-height: 30px;
    }
    @media screen and (max-width: 1400px) {
        font-size: 20px;
    }
`;
