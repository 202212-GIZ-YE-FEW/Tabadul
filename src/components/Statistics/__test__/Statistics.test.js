import renderer from "react-test-renderer";

import Statistics from "../Statistics";

it("renders correctly", () => {
    const tree = renderer.create(<Statistics />).toJSON();
    expect(tree).toMatchSnapshot();
});
