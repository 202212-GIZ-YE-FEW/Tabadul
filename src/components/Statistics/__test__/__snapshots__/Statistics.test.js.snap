// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="css-y3q6ji"
>
  <img
    alt="StatisticsContainerImage"
    className="css-1u60kn0"
    data-nimg="1"
    decoding="async"
    height={464.69}
    onError={[Function]}
    onLoad={[Function]}
    src="/_next/image?url=%2Fimg.jpg&w=2048&q=75"
    srcSet="/_next/image?url=%2Fimg.jpg&w=1080&q=75 1x, /_next/image?url=%2Fimg.jpg&w=2048&q=75 2x"
    style={
      Object {
        "color": "transparent",
      }
    }
    width={1006.7}
  />
  <div
    className="css-5mu23m"
  >
    <div
      className="css-18ul79m"
    >
      <img
        alt="StatisticImage"
        className="css-1wbd6tf"
        data-nimg="1"
        decoding="async"
        height={78.24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={78.24}
      />
      <h2
        className="css-1gystw5"
      >
        1/3
      </h2>
      <p
        className="css-3011nz"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="css-18ul79m"
    >
      <img
        alt="StatisticImage"
        data-nimg="1"
        decoding="async"
        height={78.24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={78.24}
      />
      <h2
        className="css-1gystw5"
      >
        75%
      </h2>
      <p
        className="css-3011nz"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="css-18ul79m"
    >
      <img
        alt="StatisticImage"
        data-nimg="1"
        decoding="async"
        height={78.24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={78.24}
      />
      <h2
        className="css-1gystw5"
      >
        312
      </h2>
      <p
        className="css-3011nz"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
    <div
      className="css-18ul79m"
    >
      <img
        alt="StatisticImage"
        data-nimg="1"
        decoding="async"
        height={78.24}
        loading="lazy"
        onError={[Function]}
        onLoad={[Function]}
        src="/_next/image?url=%2Fimg.jpg&w=256&q=75"
        srcSet="/_next/image?url=%2Fimg.jpg&w=96&q=75 1x, /_next/image?url=%2Fimg.jpg&w=256&q=75 2x"
        style={
          Object {
            "color": "transparent",
          }
        }
        width={78.24}
      />
      <h2
        className="css-1gystw5"
      >
        Fact
      </h2>
      <p
        className="css-3011nz"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      </p>
    </div>
  </div>
</div>
`;
